_minutes/2023-09-12-wecg-tpac.md

```diff
@@ -0,0 +1,242 @@
+ * [timothy] Not familiar with WebAuthn; In Safari we'd be fine with allowing extensions to be associated with the domain, because the extension is associated with an app and a domain. We go to great lengths to hide the randomized extension URL, and put effort in showing the extension name instead in cases such as permission prompts.
+ * [devlin] Are there use cases here that are not solvable or that are overly onerous/ugly. Sounds like it may be possible to do something with cross-origin iframes. If nothing else we should explore this more in order to gauge the priority. For example, if the only way to do this is to open a new tab and that's disruptive to the user experience, that's not great.
+ * [devlin] … I would almost say that you should be able to declare any origin as long as you have host permissions to it. Because that's true today.
+ * [carlos] This would also be useful when importing extensions between browsers (e.g. Firefox has recently introduced a feature to import extensions from Chrome) and there might be more use cases for associations.
+ * [devlin] That's interesting. I hadn't considered that. A similar thing we've discussed is not having a host permission confirmation if you can prove you own the domain. For example, if Dashlane can prove they own the extension and domain, then the extension should be able to modify the site because of course it can.
+ * [tim cappalli] For clarity, what is a host permission in this context?
+ * [devlin] Security boundary in extensions that allows extensions to inject scripts, modify the page, extract content, basically get full access to a set.
+ * [tomislav] It can also call navigator.credentials in the context of the web page.
+ * [devlin] To me, the existing ability to do so means that it would be misleading if we were to suggest that only one specific extension is in control of the credentials. I don't see any meaningful security benefit from supporting this RP ID in WebAuth.
+ * [??? guy in the back] I'm not too familiar - can the mechanism as described here do that?
+ * [nick] (introduces himself as 1Password representative) Yes we can do that.
+ * [rew] Maybe I'm missing some of the details, but wouldn't the browser be fetching this data from the domain, not the extension?
+ * [devlin] Yes. But me, a malicious extension, can inject whatever I want on a website after the proper authentication has completed. A script can reach in and extract the data it wants from the page.
+ * [rew] Doesn't that break the WebAuthN security model?
+ * [devlin] Yes
+ * [nick] You can't intercept the signing event, but you can intercept the response coming out which is a valid attack.
+ * [carlos] Wouldn't that mean that it's more secure for an extension to perform this work because the extension would get the PR ID rather than a website? Since the PR ID would only be accessible by extensions and linked extensions. While a PR ID from a website could be received by extension content scripts.
```

Comment:
```suggestion
 * [carlos] Wouldn't that mean that it's more secure for an extension to perform this work because the extension would get the RP ID rather than a website? Since the RP ID would only be accessible by extensions and linked extensions. While a RP ID from a website could be received by extension content scripts.
```